src/logging.cpp

```diff
@@ -51,8 +84,43 @@ bool BCLog::Logger::StartLogging()
         assert(!m_file_path.empty());
         m_fileout = fsbridge::fopen(m_file_path, "a");
         if (!m_fileout) {
-            return false;
+            return strprintf("Could not open debug log file %s", m_file_path.string());
+        }
+
+        // Special files (e.g. device nodes) may not have a size.
+        try {
+            m_file_size = fs::file_size(m_file_path);
+        } catch (const fs::filesystem_error&) {
+            // We can't do log rotation if it's not a regular file.
+            if (m_rotate_requested) {
+                return strprintf(
+                    "The debug log file %s does not appear to be a regular file; "
+                    "this prevents log rotation. Please disable log rotation by "
+                    "specifying -debuglogrotatekeep=0 or specify a regular file "
+                    "as the debug log file using -debuglogfile.",
+                    m_file_path.string());
+            }
+            m_rotate_keep = 0;
+        }
+        // It gets complicated to add a timestamp to an arbitrary file name,
+        // so require "debug.log".
+        if (m_file_path.filename() != "debug.log") {
+            if (m_rotate_requested) {
+                return strprintf(
+                    "The debug log file %s is not named \"debug.log\"; "
+                    "this prevents log rotation. Please disable log rotation by "
+                    "specifying -debuglogrotatekeep=0 or specify a debug log file "
+                    "path that ends in \"/debug.log\" using -debuglogfile.",
+                    m_file_path.string());
```

Comment:
this output isn't showing (not sure why). i ran:

```
./src/bitcoind -daemon -debuglogrotatekeep=3 -debugloglimit=1 -debuglogfile=/home/josibake/bitcoin/j.log
```
and got "Bitcoin Core Starting," but it fails to start without any errors.

Reply:
I think that's because you've started it with `-daemon`. It hasn't initialized the logging yet, so it has nowhere to write the error message. If you start without `-daemon` you should see the error.

Reply:
makes sense. i do think its important to throw an error when `-daemon` is used, else the user is likely to assume there node has started fine after seeing "Bitcoin Core Starting." for example, if i try to start my node with:

```bash
./src/bitcoind -datadir=foo -daemon
```
i will get:
```
Error: Specified data directory "foo" does not exist.
```

Reply:
@josibake 
> i do think its important to throw an error when -daemon is used ...

I looked into this more, and it's not trivial to fix. The problem is that bitcoind has already "daemonized" itself by the time it gets around to initializing the logging, which is later than when it tries to open the data directory. This problem happens without the current PR:
```
$ src/bitcoind -daemon -debuglogfile=some/non-existent/path
Bitcoin Core starting
$ 
```
and there's no indication that anything went wrong (the exit status is even 0, success). So I think this can be fixed in a separate PR.